src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,591 @@ title: Builder.io & Astro
+
+The `<script>` tag here won't impact your site's performance, since nobody will visit it, except for developers.   
+
+#### Setting the new route as the preview URL
+
+1. Copy the full URL of your preview to your clipboard. It should include the protocol and go something like this: `https://{your host}/builder-preview`. One way to do this is to run your Astro project (e.g. start the dev server),  type in the route into the address bar of the browser, then click into the address bar again to show the protocol.
+
+2. Go to the **Models** tab in your Builder space, pick the model  you've created and paste the URL from step 1 into the **Preview URL** field. Make sure the URL is complete and includes the protocol, for example `https://`.
+
+3. Click the **Save** button in the upper right. 
+
+:::tip
+When you deploy your site, change the preview URL to match your production URL, for example `https://myAwesomeAstroBlog.com/builder-preview`.
+:::
+
+#### Testing the preview URL set up 
+1. Make sure your site is live (e.g. your dev server is running) and the `/builder-preview` route is working.
+
+2. In your Builder space under the **Content** tab, click on **+ New** to create a new content entry for your `blogpost` model.
+
+3. In the Builder editor that just opened, you should be able to see the `builder-preview.astro` page with a big inviting **+ Add Block** in the middle!
+
+:::tip[Troubleshooting]
+
+Things can sometimes go wrong when setting up the preview. If something's not right, you can try one of these things:
+
+* Make sure the site is live - for example, your dev server is running.
+
+* Make sure that the URLs match exactly - the one in your Astro project and the one set in the Builder app.
+
+* Make sure it's the full URL including the protocol, for example `https://`.
+
+* If you're working in a virtual environment like Stackblitz or Gitpod, you might have to copy and paste the URL again when you restart your workspace, since this usually generates a new URL for your project.
+
+For more ideas, read (Builder's troubleshooting guide)[https://www.builder.io/c/docs/guides/preview-url-working].
+:::
+
+### Creating a blog post
+
+:::tip[For visual types]
+You can find a short video showing this process in one of Builder's official tutorials [here](https://www.builder.io/blog/creating-blog#create-a-blog-article).
+:::
+
+1. Before you can create new content in Builder, make sure [the preview URL for your model is working](#setting-up-the-preview-url). 
+
+2. Open the visual editor for a content entry of our `blogpost` model. If you've followed [the previous section](#testing-the-preview-url-set-up), you can use that entry. Otherwise create a new one: Go to the **Content** tab in your Builder space and click on **+ New**. 
```

Comment:
```suggestion
2. Open the visual editor for a content entry of our `blogpost` model. If you've followed [the previous section](#testing-the-preview-url-set-up), you can use that entry. Otherwise create a new one: Go to the **Content** tab in your Builder space and click on **New**. 
```